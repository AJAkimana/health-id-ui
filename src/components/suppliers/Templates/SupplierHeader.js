import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import BackIcon from '@material-ui/icons/ArrowBack';
import { Link } from 'react-router-dom';
import {
  Typography, Grid, Tooltip, IconButton
} from '@material-ui/core';

const SupplierHeader = (props) => {
  const {
    classes, previousPage, supplier, children
  } = props;
  return (
    <Fragment>
      <Grid container direction="row" alignItems="center" className={classes.arrowButtonGrid}>
        <Grid item>
          <Link to={previousPage}>
            <Tooltip title="Back To Suppliers">
              <IconButton>
                <BackIcon className={classes.arrowIcon} />
              </IconButton>
            </Tooltip>
          </Link>
        </Grid>
        <Grid item>
          <Typography variant="h5" className={classes.arrowButtonLabel}>
            {supplier.name}
          </Typography>
        </Grid>
        <Grid className={classes.buttonGrid}>{children}</Grid>
      </Grid>
    </Fragment>
  );
};
SupplierHeader.propTypes = {
  supplier: PropTypes.shape({
    name: PropTypes.string
  }).isRequired,
  previousPage: PropTypes.string.isRequired,
  classes: PropTypes.instanceOf(Object).isRequired,
  // eslint-disable-next-line react/require-default-props
  children: PropTypes.node,
};

export default SupplierHeader;
