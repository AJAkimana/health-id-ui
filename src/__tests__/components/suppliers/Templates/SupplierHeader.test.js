import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SupplierHeader from '../../../../components/suppliers/Templates/SupplierHeader';

const props = {
  supplier: {
    isApproved: false,
    id: 'AlphaX',
    address: '​​​​​NNPC Towers, Central Business District, Abuja.',
    name: 'Damian Inc',
    mobile: '08137519698',
    rating: '5',
    image:
      'https://justcreative.com/wp-content/uploads/2019/09/brand-strategy-workbook-1.jpg',
    email: 'test@gmail.com',
    tier: 'Manufacturer',
    paymentTerms: 'Daily',
    creditDays: '3 days',
  },
  renderTextField: jest.fn(),
  variables: {},
  classes: {},
  session: {
    me: {
      role: {
        name: 'Master Admin'
      },
      activeOutlet: {
        outletpreference: {
          outletTimezone: {
            name: "Africa/Nairobi"
          }
        }
      }
    }
  }
};

describe('SupplierHeader', () => {
  it('should render the SupplierHeader component properly', () => {
    const setupSupplierHeader = () => {
      const wrapper = shallow(<SupplierHeader {...props} />)
      return wrapper
    }

    let wrapper;
    wrapper = setupSupplierHeader();
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
