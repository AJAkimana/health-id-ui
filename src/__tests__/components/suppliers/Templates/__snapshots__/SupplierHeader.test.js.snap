// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierHeader should render the SupplierHeader component properly 1`] = `
<Fragment>
  <WithStyles(Grid)
    alignItems="center"
    container={true}
    direction="row"
  >
    <WithStyles(Grid)
      item={true}
    >
      <Link>
        <WithStyles(Tooltip)
          title="Back To Suppliers"
        >
          <WithStyles(IconButton)>
            <pure(ArrowBackIcon) />
          </WithStyles(IconButton)>
        </WithStyles(Tooltip)>
      </Link>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
    >
      <WithStyles(Typography)
        variant="h5"
      >
        Damian Inc
      </WithStyles(Typography)>
    </WithStyles(Grid)>
    <WithStyles(Grid) />
  </WithStyles(Grid)>
</Fragment>
`;
